core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java

```diff
@@ -264,15 +394,28 @@ public void setup(Blackhole blackhole)
             // We don't check blackhole is not null, because blackhole has to be injected by jmh (should not be created manually)
             // and in case of unit test it will be null
             this.blackhole = blackhole;
-            types = nCopies(channelCount, type.getType());
-            dataPage = type.createPage(types, POSITION_COUNT, nullRate);
+            types = type.getTypes(channelCount);
+            dataPage = type.createPage(types, positionCount, nullRate);
             pageCount = type.getPageCount();
+            nullChannel = type.getNullChannel();
             types = ImmutableList.<Type>builder()
                     .addAll(types)
                     .add(BIGINT) // dataPage has pre-computed hash block at the last channel
                     .build();
         }
 
+        private static Page page(int positionCount, int channelCount, Supplier<Block> standardBlock, Block partitionBlock)
+        {
+            ImmutableList.Builder<Block> blocks = ImmutableList.builder();
+            for (int i = 0; i < channelCount; i++) {
+                blocks.add(standardBlock.get());
+            }
+            blocks.add(partitionBlock);
+            return new Page(
```

Comment:
Hash should be precalculated too. We rather don't want to benchmark interpreted hash generator since it will shadow perf of partition output operator itself

Reply:
I may be missing something but here we only create a page with given `partitionBlock` as the last block, which is gonna be configured in the operator as (precomputed) partition/hash channel.